materials/views.py

```diff
@@ -1,32 +1,59 @@
 from rest_framework import generics, viewsets
+from rest_framework.permissions import IsAuthenticated
 
 from materials.models import Course, Lesson
 from materials.serializers import CourseSerializer, LessonSerializer
+from users.permissions import IsModer, IsOwner
 
 
 class CourseViewSet(viewsets.ModelViewSet):
     serializer_class = CourseSerializer
     queryset = Course.objects.all()
+    permission_classes = [IsAuthenticated]
+
+    def perform_create(self, serializer):
+        course = serializer.save()
```

Comment:
можно обойтись одним запросов в БД (сейчас ты делаешь два - INSERT + UPDATE)
```suggestion
        serializer.save(owner=self.request.user)
```
С уроком так же

Reply:
Оптимизировал запрос в БД